### PR TITLE
DCOS_OSS-5511: added dcos_install.ps1, install_choco.ps1

### PR DESCRIPTION
**Pull request by sergiimatusEPAM**
## High-level description

There are 2 Powershell scripts added, which then will executed by Ansible and then playing key role for installing Windows Agent. In general they install pre-requisites, and execute Winpanda setup on Windows agents

## Corresponding DC/OS tickets (required)

  - [DCOS_OSS-5511](https://jira.mesosphere.com/browse/DCOS_OSS-5511) Develop install.ps1 script for Winpanda.

## Checklist for component/package updates:

@DominikDary , @jongiddy , adding to review of PR

### CODE REVIEW

File: gen/build_deploy/powershell/dcos_install.ps1 [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        Register-ScheduledTask -TaskName 'RunOnce' -TaskPath '\CustomTasks\' -Action $action -Trigger $trigger -Principal $principal -Description 'Winpanda task to run once after winagent provision.' -ErrorAction Stop
+    } else {
+        Write-Log("$($dir)\RunOnce.ps1 already exists. Skipping")
+    }
+}
+
+function SetupMesosDNS($masters_ip) {
+    Write-Log("Mesos-DNS setup with following $masters_ip")
+    #TO DO: Mesos-DNS logic to be placed here
+}
```
Comment (DominikDary) [inline gen/build_deploy/powershell/dcos_install.ps1]: what is the purpose of this method? I understand that this is needed on the long run, but what is the purpose to have it in the first iteration?

File: gen/build_deploy/powershell/install_choco.ps1 [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+# UPDATE THIS PATH
+$localChocolateyPackageFilePath = 'c:\dcos\chocolatey_offline   \chocolatey.0.10.15.nupkg'
+# Determine unzipping method
+# 7zip is the most compatible, but you need an internally hosted 7za.exe.
+# Make sure the version matches for the arguments as well.
+# Built-in does not work with Server Core, but if you have PowerShell 5
+# it uses Expand-Archive instead of COM
+$unzipMethod = 'builtin'
+#$unzipMethod = '7zip'
+#$7zipUrl = 'https://chocolatey.org/7za.exe' (download this file, host internally, and update this to internal)
```
Comment (DominikDary) [inline gen/build_deploy/powershell/install_choco.ps1]: what is the purpose of this lines? They can be removed, right?

File: gen/build_deploy/powershell/dcos_install.ps1 (hunk 1 above)
Comment (sergiimatusEPAM) [reply gen/build_deploy/powershell/dcos_install.ps1]: correct, for now this is down by Ansible in Alpha and in Beta Sprint 1, but it needs to be done in dcos_install.ps1 to simplify offline setup/pre-setup

File: gen/build_deploy/powershell/dcos_install.ps1 [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+# function GetDestination([String] $version, [String] $destfile) {
+#     $versiondir = "$($basedir)\$($version)"
+#     if (-not (test-path "$versiondir") ) {
+#         Write-Log("$($versiondir) folder doesn't exist", "Error")
+#     } else {
+#         $output = "$($versiondir )\$($destfile)"
+#         Write-Log("Placing downloaded file as $output")
+#     }
+#     return $output
+# }
```
[the remark this answers is not in the document]
Comment (sergiimatusEPAM) [reply gen/build_deploy/powershell/dcos_install.ps1]: done

File: gen/build_deploy/powershell/dcos_install.ps1 [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+function Download([String] $url, [String] $file) {
+    # Download url
+    # TO DO: remove
+    # DEPRECATED since version is not used on winagent
+    # $output = $(GetDestination $version $file)
```
[the remark this answers is not in the document]
Comment (sergiimatusEPAM) [reply gen/build_deploy/powershell/dcos_install.ps1]: done

File: gen/build_deploy/powershell/install_choco.ps1 (hunk 2 above)
Comment (sergiimatusEPAM) [reply gen/build_deploy/powershell/install_choco.ps1]: Correct. 
install_choco.ps1 is officially published and available by chocolatey.org, however we are using this for offline installation.

**Comment by DominikDary** [comment]
@mesosphere-mergebot sync

**Comment by sergiimatusEPAM** [comment]
@mesosphere-mergebot changelog-not-required Preparation to add Winpanda

**Comment by DominikDary** [comment]
@mesosphere-mergebot merge-it

**Comment by DominikDary** [comment]
@mesosphere-mergebot sync

**Comment by DominikDary** [comment]
@mesosphere-mergebot label Ship It

**Comment by jongiddy** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by DominikDary** [comment]
@orsenthil can you please help getting this merged?

**Comment by AliakseiKhomchanka** [comment]
@mesosphere-mergebot sync

**Pull request merged**